ws/src/speech/scripts/Hear.py

```diff
@@ -1,85 +1,66 @@
+            rospy.logerr(f"Code: {e.code()}")
+            raise
 
-publisher_azure = None
-publisher_whisper = None
+def callback_audio(data, args):
+    client, publisher = args
+    rospy.loginfo("Received audio data, sending to Whisper gRPC server...")
 
-def callback_azure(data):
-    # Change Sample Rate.
-    resample=SpeechApiUtils.resample_ratecv(data.data, 16000, 16000)
-    # getAllSamples.
-    allsamples=SpeechApiUtils.get_all_samples(resample[0])
-    # Publish.
-    publisher_azure.publish(allsamples)
-    rospy.loginfo("Sent to Azure node.")
-  
+    # Resample the audio to 16 kHz (if needed)
+    resampled = SpeechApiUtils.resample_ratecv(data.data, 16000, 16000)
+    all_samples = SpeechApiUtils.get_all_samples(resampled[0])
     
-def callback_whisper(data):
-    # Whisper resamples audio to 16 kHz
-    resample=SpeechApiUtils.resample_ratecv(data.data, 16000, 16000)
-    # getAllSamples.
-    allsamples=SpeechApiUtils.get_all_samples(resample[0])
-    # Publish.
-    publisher_whisper.publish(allsamples)
-    rospy.loginfo("Sent to Whisper node.")
-
+    # Convert samples to bytes
+    audio_bytes = bytes(all_samples)  # Convert the list to bytes
+    
+    # Send audio to Whisper gRPC server
+    try:
+        transcription = client.transcribe(audio_bytes)
+        rospy.loginfo(f"Transcription received: {transcription}")
 
-def both_callback(data):
-    rospy.loginfo("*Received a voice audio, computing...*")
-    if (FORCE_ENGINE == "online" or 
-        (FORCE_ENGINE == "none" and SpeechApiUtils.is_connected())):
-        callback_azure(data)
-    else:
-        callback_whisper(data)
+        # Publish the transcription to a ROS topic
+        msg = String()
+        msg.data = transcription
+        publisher.publish(msg)
+        rospy.loginfo("Transcription published to ROS topic.")
+    except grpc.RpcError as e:
+        rospy.logerr(f"gRPC error: {e}")
 
 def main():
-    # In ROS, nodes are uniquely named. If two nodes with the same
-    # name are launched, the previous one is kicked off. The
-    # anonymous=True flag means that rospy will choose a unique
-    # name for our 'listener' node so that multiple listeners can
-    # run simultaneously.
     rospy.init_node('hear', anonymous=True)
     rospy.loginfo("*Starting Hear Node*")
 
-    global FORCE_ENGINE
-    FORCE_ENGINE=rospy.get_param('~FORCE_ENGINE', 'online')
+    # Create the gRPC client
+    whisper_client = WhisperClient("127.0.0.1:50051")
 
-    global publisher_azure, publisher_whisper
-    # For publishing when online to Azure node to it compute it and publish.
-    publisher_azure = rospy.Publisher('UsefulAudioAzure', AudioData, queue_size=5)
-    
-    # For publishing when online to Whispe node to compute it and publish.
-    publisher_whisper = rospy.Publisher('UsefulAudioWhisper', AudioData, queue_size=5)
+    # ROS Publisher for transcription
+    publisher = rospy.Publisher("/speech/transcription", String, queue_size=10)
```

Comment:
Modify publisher to use the previous topic (`/speech/raw_command`) to avoid modifying the other files.